.github/workflows/_hack_make.yml

```diff
@@ -28,6 +28,9 @@ jobs:
         with:
           go-version: "1.21"
           cache-dependency-path: "internal/mage/go.sum"
+      - name: Install dagger
```

Comment:
Not a blocker but can we use https://github.com/dagger/dagger-for-github for this?

Reply:
This is tricky - because we're doing `./hack/dev` to the nested engine, it's not super trivial.

I'll have a think about this, and try and do a follow-up, but this also is entertwined with removing mage in general.

Reply:
All of this is going away in the CI setup.

Once we declare the Dagger version that we want - e.g. `dagger-v0-10-3` - the runner is provisioned with all the necessary tooling, including the Dagger CLI. The code is available in this private repository: https://github.com/dagger/dagger.io/blob/3d37ad6abaf31634aa941796ea4fe8a107e68bcd/infra/ci/eks-2023-12-15/argocd/dagger-runners/dagger-gh-runners.yaml#L164-L202

We are integrating this in https://github.com/dagger/dagger/pull/7018, currently working through a few issues.